### DISCUSSION TITLE
Where is the right file to set the environment variables?

### DISCUSSION THREAD

**Discussion by BryceWG** [open-webui/open-webui#7218] [Q&A]
I pulled the complete project files from the repository and deployed the project using docker compose, but I could not find the config.py file in ./backend/ (it's also not in the repository):
![gmh2vjgc u5d](https://github.com/user-attachments/assets/3db0af90-ece4-4437-a7af-1bd19ce72a0e)
so I modified the .env.example file in the root directory and renamed it to .env, but it didn't work, like this:
![5enk2pp0 alo](https://github.com/user-attachments/assets/b0442f7a-4505-4cac-ba50-41ff10211f07)
I am trying to reduce the memory usage of the project, but the settings in the env do not take effect. I also looked for other possible configuration files, such as config.json, but I did not find any (even with file search)
If anyone knows where to set these environment variables, please let me know

**Comment by rikkooo** [comment]
After digging for a couple of hours for this intel I fount it =>

In my case, "OI" is installed within a **conda environment, with a pip package.**
For my test system that's the BASE_DIR 
BASE_DIR = PosixPath('/root/miniconda3/envs/owui/lib/python3.11')  
**owui** in this case is my environment

and it loads the '.env' file from the root of the BASE_DIR => load_dotenv(find_dotenv(str(BASE_DIR / ".env")))

I've already tested and booom, it's working. Just place your .env there (BASE_DIR) and it will read and update the 'OI' variables!

PS. To check the root of Conda environments, run 'conda list'.

**Reply by scalenow** [reply]
@rikkooo followed your instructions and could not find the .env file .(openweb3.11) root@scalenowai:~/miniconda3/envs/openweb3.11# .I guess my file should be  openweb3.11 ,but there is none.Please advise

**Reply by navti** [reply]
@scalenow  I believe 'openweb3.11' is the environment name? If yes, there will be a 'lib' folder inside it and then a python3.xx folder inside the lib folder. You can create a new .env file inside the python3.xx folder and put all the environment variables inside that.

**Reply by scalenow** [reply]
@navti Thanks

**Comment by scalenow** [comment]
@navti  any pointers to this issue https://github.com/open-webui/open-webui/issues/9175

**Reply by navti** [reply]
@scalenow After searching a bit on the internet, looks like ollama doesn't have support for SSL. Check this [reddit link](https://www.reddit.com/r/ollama/comments/1gvtv7a/how_to_run_ollama_with_ssl/). Someone had also opened a [PR #1310](https://github.com/ollama/ollama/pull/1310) to add SSL support but it never got merged.

**Reply by scalenow** [reply]
@navti ssl is working no issue ,but on https cannot connect to ollama and get reesponce, Able to connect to ollama with http

**Reply by navti** [reply]
@scalenow Are you sure that ollama supports https endpoint? Anywhere I look, it suggests that ollama supports only http endpoint. Here's one [comment](https://github.com/ollama/ollama/pull/1310#issuecomment-2490286537) from one of the ollama's lead contributors. Quoting them below

> Hi all, so sorry for the slow reply. Thanks @rootedbox for creating this PR. Ollama was designed to provide an http endpoint that can be fronted by other proxies and load balancers that serve requests over TLS (including certificate rotation, etc). Some tools that work well for this are `nginx` or `caddy`. Let me know if that might not work and we can revisit.

**Comment by scalenow** [comment]
@navti please comment on the following link,2 seperate topics  https://github.com/open-webui/open-webui/discussions/9176

**Comment by scalenow** [comment]
@navti This has been addressed,thanks for your help
